src/mmw/js/src/draw/views.js

```diff
@@ -64,7 +65,13 @@ function validateShape(polygon) {
     var area = coreUtils.changeOfAreaUnits(turfArea(polygon), 'm<sup>2</sup>', 'km<sup>2</sup>'),
         d = new $.Deferred();
 
-    if (area > MAX_AREA) {
+    var selfIntersectingShape = turfKinks(polygon).features.length > 0;
+
+    if (selfIntersectingShape) {
```

Comment:
RWD can produce this kind of polygon as well, based on your previous testing right?  If so, we may want to apply this validation to both paths.